crates/common/upload/src/upload.rs

```diff
@@ -27,26 +29,69 @@ fn default_backoff() -> ExponentialBackoff {
     }
 }
 
+/// Auto tries to detect the mime from the given file extension without direct file access.
+/// Custom sets a custom Content-Type.
+/// If multipart request is needed, choose FormData.
 #[derive(Debug, Clone, Eq, PartialEq)]
 pub enum ContentType {
+    Auto,
+    Custom(String),
+    FormData(FormData),
+}
+
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub enum Mime {
     TextPlain,
     ApplicationOctetStream,
 }
```

Comment:
Why not using [`mine::Mine`](https://docs.rs/mime/latest/mime/#mime)?

Reply:
I was hesitated to add `mime` crate to `Cargo.toml`. However, as you suggested, we can replace it and thin-edge itself has already dependency on the `mime` crate because of `reqwest`.

Reply:
Done with 7b511cd60680348c0fda66e26b5979cf05fb50df